### PR TITLE
Update Kafka configuration types

### PR DESCRIPTION
**Pull request by gotson**
## Hey, I just made a Pull Request!

According to https://nodejs.org/dist/latest-v8.x/docs/api/tls.html#tls_tls_createsecurecontext_options the `ssl` options `ca`, `key`, and `cert` are all optional. In my case, i need to pass `rejectUnauthorized`, but I can't do so as the backend expects a `key`. Passing an empty string works, but everything becomes redacted in my log because of that.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/fresh-zebras-hug.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+"@backstage/plugin-kafka-backend": patch
+"@backstage/plugin-kafka": patch
```
Comment (freben) [inline .changeset/fresh-zebras-hug.md:3]: Doesn't seem like there was any change in this package.

**Review by freben — APPROVED** [review]
Looks good (seems to match the underlying types correctly), except you need to remove that one extra line in the changeset. Then should be ready to merge!

File: .changeset/fresh-zebras-hug.md (hunk 1 above)
Comment (freben) [reply .changeset/fresh-zebras-hug.md:3]: Oh and it wants these to all be single quotes, not double

**Review by freben — APPROVED** [review]

**Comment by suuus** [comment]
Thanks for your contribution @gotson 🙌 Please let me know if you’d like some help to contribute further.

**Pull request merged**